sql-plugin/src/main/scala/org/apache/spark/sql/rapids/arithmetic.scala

```diff
@@ -72,6 +74,23 @@ case class GpuUnaryMinus(child: Expression, failOnError: Boolean) extends GpuUna
       // Because of 2s compliment we need to only worry about the min value for integer types.
       GpuAnsi.assertMinValueOverflow(input, "minus")
     }
+
+    if (GpuTypeShims.isDayTimeIntervalType(dataType)) {
+      // For day time interval, Spark throws an exception when overflow,
+      // regardless of whether `SQLConf.get.ansiEnabled` is true or false
+      withResource(Scalar.fromLong(Long.MinValue)) { minVal =>
+        GpuAnsi.assertMinValueOverflowImp(minVal, input, "minus")
+      }
+    }
+
+    if (GpuTypeShims.isYearMonthIntervalType(dataType)) {
```

Comment:
I'd recommend moving this stuff into the following `dataType match` via something like this:
```
  case t if GpuTypeShims.isDayTimeIntervalType(t) => ...
  case t if GpuTypeShims.isYearMonthIntervalType(t) => ....
```

Reply:
done